app/assets/javascripts/admin/addon/controllers/admin-users-list-show.js

```diff
@@ -1,25 +1,34 @@
+import { tracked } from "@glimmer/tracking";
 import Controller from "@ember/controller";
 import { action } from "@ember/object";
-import { observes } from "@ember-decorators/object";
+import { service } from "@ember/service";
 import { computedI18n } from "discourse/lib/computed";
 import CanCheckEmails from "discourse/mixins/can-check-emails";
 import { INPUT_DELAY } from "discourse-common/config/environment";
 import discourseDebounce from "discourse-common/lib/debounce";
-import discourseComputed from "discourse-common/utils/decorators";
+import discourseComputed, { bind } from "discourse-common/utils/decorators";
 import { i18n } from "discourse-i18n";
+import BulkUserDeleteConfirmation from "admin/components/bulk-user-delete-confirmation";
 import AdminUser from "admin/models/admin-user";
 
 export default class AdminUsersListShowController extends Controller.extend(
   CanCheckEmails
 ) {
-  model = null;
+  @service dialog;
+  @service modal;
+
+  @tracked bulkSelect = false;
+  @tracked displayBulkActions = false;
+  @tracked bulkSelectedUserIdsSet = new Set();
+  @tracked bulkSelectedUsersMap = {};
```

Comment:
You might be able to use a JS `Map` here instead

Reply:
I was hoping that `Map` could replace both the `Set` and the POJO, but `Map` doesn't work with Ember's `get` method (see the link), so I have to stick to `Set` and POJO.

https://github.com/discourse/discourse/pull/29744/files#diff-dcc9c57dab8ff1ac002b005e803cf648b760828994fc97ad2b249c0a265236f6R184